ASIM/dev/Delete-SentinelFunction/Delete-LASavedFunction.ps1

```diff
@@ -0,0 +1,107 @@
+    <#
+        .SYNOPSIS
+        Deletes saved searches from Log Analytics workspace
+
+        .DESCRIPTION
+        Deletes functions from Log Analytics workspace.
+        Takes a list of strings for the function names.
+
+        .PARAMETER FunctionName
+        Specifies a comma delimited list of saved function name to be deleted. Accepts wildchars. 
+
+        .PARAMETER WorkspaceName
+        Specifies the workspace where functions are saved.
+
+        .PARAMETER ResourceGroup
+        Specifies the resource group of workspace.
+
+        
+        .PARAMETER Force
+        If set the user will not be prompted for approval. Be causious!.
+
+        .PARAMETER Category
+        Specifies the category of deleted functions.
+
+        .PARAMETER Category
+        Specifies which subscription (aka context) to use.
+
+
+        .EXAMPLE
+        PS> Delete-LASavedFunction DeleteM*, TestFunction* -Subscription "Contoso Production" -Workspace contosoc_ws -ResourceGroup soc_rg
+
+        .LINK
+        https://github.com/Azure/Azure-Sentinel/tree/master/ASIM/dev/Delete-LASavedFunction
+    #>
+[CmdletBinding(PositionalBinding=$false)]
+param(
+    [Parameter(Mandatory=$true ,HelpMessage="Comma delimited function names to delete.", Position=0 )]
+    [SupportsWildcards()]
+    [string[]]$FunctionName,
+
+    [Parameter(Mandatory=$true)]
+    [string]$WorkspaceName,
+
+    [Parameter(Mandatory=$true)]
+    [string]$ResourceGroup,
+
+
+    [Parameter(Mandatory=$false)]
+    [switch]$Force,
+
+    [Parameter(Mandatory=$false)]
+    [string]$Category,
+
+    [Parameter(Mandatory=$false)]
+    [string]$Subscription
+)
+
+Import-Module Az.OperationalInsights
+
+if ((Get-AzContext) -eq $null)
+{
+   Connect-AzAccount
+}
+
+$default_subscription = (Get-AzContext).Subscription.Name
+if ($PSBoundParameters.ContainsKey("Subscription") )
+{
+    $context = $Subscription
+    Write-Host "Running on subscription: $context."
+}
+else
+{
+    $context = $default_subscription
+    Write-Host "Running on default subscription: $context. To run on a different subscription use parameter -Subscription"
+}
+if ($context -ne $default_subscription){
+    $suppress=Set-AzContext -Subscription $context
+}
+
+$yestoall=$PSBoundParameters.ContainsKey('Force')
```

Comment:
Is Force the standard in powershell? I think there is a standard name for this parameter.

Reply:
It is used: [Remove-Item -Force](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/remove-item?view=powershell-7.2)